File: mifosng-android/src/main/java/com/mifos/mifosxdroid/passcode/PassCodeActivity.java
package com.mifos.mifosxdroid.passcode;

import android.content.Intent;
import android.view.View;

import com.mifos.mifosxdroid.R;
import com.mifos.mifosxdroid.SplashScreenActivity;
import com.mifos.mifosxdroid.core.util.Toaster;
import com.mifos.mifosxdroid.online.DashboardActivity;
import com.mifos.mobile.passcode.MifosPassCodeActivity;
import com.mifos.mobile.passcode.utils.EncryptionUtil;

public class PassCodeActivity extends MifosPassCodeActivity {

    @Override
    public int getLogo() {
        return R.drawable.mifos_logo;
    }

    @Override
    public void startNextActivity() {
        startActivity(new Intent(this, DashboardActivity.class));
    }

    @Override
    public void startLoginActivity() {
        Intent i = new Intent(PassCodeActivity.this, SplashScreenActivity.class);
        i.setFlags(Intent.FLAG_ACTIVITY_NEW_TASK | Intent.FLAG_ACTIVITY_CLEAR_TASK);
        startActivity(i);
        finish();
    }

    @Override
    public void showToaster(View view, int msg) {
        Toaster.show(view, msg);
    }

    @Override
    public int getEncryptionType() {
        return EncryptionUtil.ANDROID_CLIENT;
    }

}

File: mifosng-android/src/main/java/com/mifos/mifosxdroid/passcode/PassCodeActivity.kt
package com.mifos.mifosxdroid.passcode

import android.os.Bundle
import androidx.appcompat.app.AppCompatActivity
import com.mifos.utils.Constants

class PassCodeActivity : AppCompatActivity() {

    private var currPassCode: String? = null
    private var isToUpdatePassCode: Boolean = false

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
//        findViewById<NestedScrollView>(com.mifos.mobile.passcode.R.id.cl_rootview).setBackgroundColor(
//            android.R.attr.colorBackground
//        )

        intent?.let {
            currPassCode = it.getStringExtra(Constants.CURR_PASSWORD)
            isToUpdatePassCode = it.getBooleanExtra(Constants.IS_TO_UPDATE_PASS_CODE, false)
        }
    }

//    override fun showToaster(view: View?, msg: Int) {
//        Toaster.show(view, msg, Toaster.SHORT)
//    }
//
//    override fun startLoginActivity() {
//        startActivity(Intent(this, AndroidClientActivity::class.java))
//        finish()
//    }
//
//    override fun getLogo(): Int {
//        return R.drawable.mifos_logo
//    }
//
//    override fun getEncryptionType(): Int {
//        return EncryptionUtil.FINERACT_CN
//    }
//
//    override fun startNextActivity() {
//        startActivity(Intent(this, AndroidClientActivity::class.java))
//    }
//
//    override fun onBackPressed() {
//        super.onBackPressed()
//        if (isToUpdatePassCode && !currPassCode.isNullOrEmpty()) {
//            PasscodePreferencesHelper(this).apply {
//                savePassCode(currPassCode)
//            }
//        }
//        finish()
//    }
}
